src/server.rs

```diff
@@ -1607,4 +1699,2421 @@ errorCounts
+        };
+
+        let result = block_on(server.completion(params.clone()))
+            .unwrap()
+            .unwrap();
+
+        let items = match result.clone() {
+            lsp::CompletionResponse::List(l) => l.items,
+            _ => unreachable!(),
+        };
+
+        let labels: Vec<&str> =
+            items.iter().map(|item| item.label.as_str()).collect();
+
+        let expected = vec!["csv", "file", "mode", "url"];
+
+        assert_eq!(expected, labels);
+    }
+
+    #[test]
+    fn test_options_completion() {
+        let fluxscript = r#"import "strings"
+import "csv"
+
+cal = 10
+env = "prod01-us-west-2"
+
+cool = (a) => a + 1
+
+option task = {
+  name: "foo",        // Name is required.
+  every: 1h,          // Task should be run at this interval.
+  delay: 10m,         // Delay scheduling this task by this duration.
+  cron: "0 2 * * *",  // Cron is a more sophisticated way to schedule. 'every' and 'cron' are mutually exclusive.
+  retry: 5,           // Number of times to retry a failed query.
+}
+
+newNow = t
+
+errorCounts = from(bucket:"kube-infra/monthly")
+    |> range(start: -3d )
+    |> filter(fn: (r) => r._measurement == "query_log" and
+                         r.error != "" and
+                         r._field == "responseSize" and
+                         r.env == env)
+    |> group(columns:["env", "error"])
+    |> count()
+    |> group(columns:["env", "_stop", "_start"])
+
+errorCounts
+    |> filter(fn: (r) => strings.containsStr(v: r.error, substr: "AppendMappedRecordWithNulls"))
+
+"#;
+        let server = create_server();
+        open_file(&server, fluxscript.to_string());
+
+        let params = lsp::CompletionParams {
+            text_document_position: lsp::TextDocumentPositionParams {
+                text_document: lsp::TextDocumentIdentifier {
+                    uri: lsp::Url::parse(
+                        "file:///home/user/file.flux",
+                    )
+                    .unwrap(),
+                },
+                position: lsp::Position {
+                    line: 16,
+                    character: 10,
+                },
+            },
+            work_done_progress_params: lsp::WorkDoneProgressParams {
+                work_done_token: None,
+            },
+            partial_result_params: lsp::PartialResultParams {
+                partial_result_token: None,
+            },
+            context: Some(lsp::CompletionContext {
+                trigger_kind: lsp::CompletionTriggerKind::Invoked,
+                trigger_character: None,
+            }),
+        };
+
+        let result = block_on(server.completion(params.clone()))
+            .unwrap()
+            .unwrap();
+
+        let items = match result.clone() {
+            lsp::CompletionResponse::List(l) => l.items,
+            _ => unreachable!(),
+        };
+
+        assert_eq!(123, items.len());
+        assert_eq!("task (self)", items.last().unwrap().label);
+    }
+}
+
+#[derive(Debug)]
```

Comment:
Is this all existing code that just had the `Cache` argument removed from signatures? If so, I'm not going to bother reviewing it.

Reply:
Yes, the vast majority of it is repurposed versions of existing code that just had `Cache` and `RequestContext` taken out of the picture. I don't think there's a need for a detailed review on anything but the `completion()` method and the added tests.